File: Data-Structures/Trees/0145_binary_tree_postorder.py
#tree

from typing import Optional 

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        #left -> parent -> child 
        res = []
        def dfs(node):
            if not node: 
                return 
            dfs(node.left)
            dfs(node.right)
            res.append(node.val)

        dfs(root)
        return res
        

File: Data-Structures/Trees/0145_binary_tree_postorder.c
// Practice for: 0145_binary_tree_postoder.py
// Bridge logic from Python to C pointers here.

#include <stdio.h>
#include <stdlib.h>

struct TreeNode {
    int val;
    struct TreeNode* left;
    struct TreeNode* right;
};

void dfs(struct TreeNode *node, int* res, int *returnSize){
    if (node == NULL){
        return;
    }

    dfs(node ->left, res, returnSize);
    dfs(node -> right, res, returnSize);
    res[(*returnSize)++] = node->val;
}

int* postorderTraversal(struct TreeNode *root, int* returnSize){
    int* res = (int*)malloc(100 * sizeof(int));
    *returnSize = 0;
    dfs(root, res, returnSize);
    return res;

}